app/models/manageiq/providers/vmware/infra_manager/refresh_parser.rb

```diff
@@ -1078,21 +1078,27 @@ def self.vm_inv_to_disk_hashes(inv, storage_uids, storage_profile_by_disk_mor =
           device_type << (backing['fileName'].nil? ? "-raw" : "-image") if device_type == 'cdrom'
 
           controller = inv.detect { |d| d['key'] == device['controllerKey'] }
-          controller_type = case controller.xsiType
-                            when /IDE/ then 'ide'
-                            when /SIO/ then 'sio'
-                            else 'scsi'
-                            end
+          controller_type =
+            if controller
+              case controller.xsiType
+              when /IDE/ then 'ide'
+              when /SIO/ then 'sio'
+              else 'scsi'
+              end
+            else
+              'unknown'
+            end
+
 
           storage_mor = backing['datastore']
 
           new_result = {
             :device_name     => device.fetch_path('deviceInfo', 'label'),
             :device_type     => device_type,
-            :controller_type => controller_type,
+            :controller_type => controller_type || 'unknown',
```

Comment:
Since you already have `if controller else 'unknown'` is the `|| 'unknown'` necessary?